src/Microsoft.DotNet.VersionTools/Automation/PullRequestCreator.cs

```diff
@@ -0,0 +1,201 @@
+
+                        if (blockedReason == null)
+                        {
+                            originBranch = new GitHubBranch(
+                                pullRequestToUpdate.Head.Ref,
+                                originBranch.Project);
+                        }
+                        else
+                        {
+                            string comment =
+                                $"Couldn't update this pull request: {blockedReason}\n" +
+                                $"Would have applied '{commitMessage}'";
+
+                            await client.PostCommentAsync(Upstream, pullRequestToUpdate.Number, comment);
+                            return;
+                        }
+                    }
+                }
+
+                bool forcePush = pullRequestToUpdate != null;
+
+                PushNewCommit(
+                    originBranch,
+                    commitMessage,
+                    forcePush);
+
+                if (forcePush)
+                {
+                    await client.UpdateGitHubPullRequestAsync(Upstream, pullRequestToUpdate.Number, title, description);
+                }
+                else
+                {
+                    await client.PostGitHubPullRequestAsync(title, description, originBranch, UpstreamBranch);
+                }
+            }
+        }
+
+        public static string NotificationString(IEnumerable<string> usernames)
+        {
+            return $"/cc @{string.Join(" @", usernames)}";
+        }
+
+        private string GetUpdateBlockedReason(GitHubClient client, GitHubPullRequest pullRequest)
+        {
+            if (pullRequest.Head.User.Login != Origin.Owner)
+            {
+                return $"Owner of head repo '{pullRequest.Head.User.Login}' is not '{Origin.Owner}'";
+            }
+            if (!pullRequest.Head.Ref.StartsWith(BranchPrefix))
+            {
+                return $"Ref name '{pullRequest.Head.Ref}' does not start with '{BranchPrefix}'";
+            }
+
+            GitCommit commit = client.GetCommitAsync(Origin, pullRequest.Head.Sha).Result;
+            if (commit.Author.Name != GitAuthorName)
+            {
+                return $"Head commit author '{commit.Author.Name}' is not '{GitAuthorName}'";
+            }
+            if (commit.Committer.Name != GitAuthorName)
+            {
+                return $"Head commit committer '{commit.Committer.Name}' is not '{GitAuthorName}'";
+            }
+            return null;
+        }
+
+        private void PushNewCommit(
+            GitHubBranch branch,
+            string commitMessage,
+            bool forcePush)
+        {
+            // Set committer in environment rather than through command start options because net45
+            // doesn't have that feature.
+            Environment.SetEnvironmentVariable("GIT_COMMITTER_NAME", GitAuthorName);
```

Comment:
I wasn't able to find arguments for committer, unfortunately. The closest I could find is running `git config ...` instead of using the env vars, but that didn't seem any better to me.

net45 here is about using `Environment.SetEnvironmentVariable` instead of setting it on the `ProcessStartInfo`, but looking back at CLI's command class, this should have been fixed with https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.Cli.Utils/Command.cs#L169-L173. Not sure why I had to remove that in the earlier PR.